scripts/globals/chocobo_digging.lua

```diff
@@ -0,0 +1,255 @@
+
+                if ((weather == 4 or weather == 5) and ItemID == 4096) then
+                  ItemID = ItemID;
+                elseif ((weather == 6 or weather == 7) and ItemID == 4097) then
+                  ItemID = ItemID;
+                elseif ((weather == 8 or weather == 9) and ItemID == 4098) then
+                  ItemID = ItemID;
+                elseif ((weather == 10 or weather == 11) and ItemID == 4099) then
+                  ItemID = ItemID;
+                elseif ((weather == 12 or weather == 13) and ItemID == 4100) then
+                  ItemID = ItemID;
+                elseif ((weather == 14 or weather == 15) and ItemID == 4101) then
+                  ItemID = ItemID;
+                elseif ((weather == 16 or weather == 17) and ItemID == 4102) then
+                  ItemID = ItemID;
+                elseif ((weather == 18 or weather == 19) and ItemID == 4103) then
+                  ItemID = ItemID;
+                else
+                  ItemID = 0;
+                end
+                
+                -- If the item is a crystal, we need to check for a double weather effect (Which results in a crystal cluster, not a single crystal)
+                if (ItemID >= 4096 or ItemID <= 4103) then
+                  if (zoneWeather == "WEATHER_DOUBLE") then
+                    ItemID = ItemID + 8;
+                  elseif (zoneWeather == "WEATHER_SINGLE") then
+                    ItemID = ItemID;
+                  else
+                    ItemID = 0;
+                  end
+                end
+                
+                -- If the item is an elemental ore, we need to check if the requirements are met
+                
+                if ((ItemID >= 1255 and ItemID <= 1262) and weather > 1 and (moon >= 10 and moon <= 40) and SkillRank >= 7) then
+                  if (ItemID == 1255 and day = 0) then
+                    ItemID = ItemID;
+                  elseif (ItemID == 1256 and day = 1) then
+                    ItemID = ItemID;
+                  elseif (ItemID == 1257 and day = 2) then
+                   ItemID = ItemID;
+                  elseif (ItemID == 1258 and day = 3) then
+                   ItemID = ItemID;
+                  elseif (ItemID == 1259 and day = 4) then
+                    ItemID = ItemID;
+                  elseif (ItemID == 1260 and day = 5) then
+                   ItemID = ItemID;
+                  elseif (ItemID == 1261 and day = 6) then
+                   ItemID = ItemID;
+                  elseif (ItemID == 1262 and day = 7) then
+                    ItemID = ItemID;
+                  else
+                    ItemID = 0;
+                end            
+            end
+            
+            -- make sure we have a valid item
+            if (ItemID ~= 0) then
+                -- make sure we have enough room for the item
+                if (player:getFreeSlotsCount() > 0) then
```

Comment:
forgot about addItem, blame every script that uses getFreeSlotsCount/addItem evar